src/DomainHost.sol

```diff
@@ -306,17 +309,47 @@ abstract contract DomainHost {
         emit Deficit(wad);
     }
 
+    /// @notice Governance is accruing surplus
+    /// @dev This is a potentially dangerous operation as a malicious domain can trigger unintended consequences due to fake surplus
+    /// @param _grain It should be the amount of grain necessary to cover debt in the guest domain [WAD]
+    function accrue(uint256 _grain) external auth returns (uint256 _wad) {
+        require(vat.live() == 1, "DomainHost/vat-not-live");
+        uint256 _dsin = dsin;
+        uint256 _ddai = ddai;
+        require(_ddai > _dsin, "DomainHost/no-surplus");
+
+        int256 diff = _int256(_grain) - _int256(grain);
+        if (diff > 0) {
+            vat.slip(ilk, address(this), diff);
+            vat.frob(ilk, address(this), address(this), address(this), diff, diff);
+            daiJoin.exit(escrow, uint256(diff));
+            grain = _grain;
+        }
+
+        unchecked { _wad = _ddai - _dsin; }
+        dai.transferFrom(address(escrow), address(this), _wad);
+        daiJoin.join(address(vow), _wad);
+
+        dsin = 0;
+        ddai = 0;
+
+        emit Accrue(_wad);
```

Comment:
There is effectively a double counting, but I'm thinking it might be better to fix it in `Guest._surplus` discounting from `sin` what we already have in `dsin` (exact same way than `Guest._deficit`). To be honest that was probably my original intention and forgot to adapt `_surplus` like I did with `_deficit`.
Maybe the end result has a problem that your solution doesn't have, but need to look at it more deeply. If both solutions work the same, I think I'd lead to prefer mine to keep the symmetry.

Reply:
Yeah you are right as well. I'm following your fix, it is probably the best, just pushing something slightly different due to the `_wad` variable.
The only annoying thing is in this example we need to start the `surplus` cycle again, as the 10 DAI end up back in the guest side. But I guess that is a problem we have due to not being able to realize the deficit instantly in the guest, so probably any solution will end up in something similar when you have parallel deficit and surplus being pushed.